src/register-event-bus.ts

```diff
@@ -0,0 +1,45 @@
+import { SEQUENTIAL_EVENT_LISTENER } from './decorators/SequentialEventListener';
+import { isEvent } from './ISequentialEventListener';
+import type { ModuleRef, ModulesContainer } from '@nestjs/core';
+import type { SequentialEventBus } from './SequentialEventBus';
+
+interface ListenerDefinition {
+  providerConstructor: any;
+  eventType: any;
+}
+
+export const registerEventBus = (modulesContainer: ModulesContainer, moduleRef: ModuleRef, eventBus: SequentialEventBus) => {
```

Comment:
I don't like, that both this and implementation in `onApplicationBootstrap` are required, since afair they are both doing the same. Or is not used?

Reply:
wow, sorry I forgot to delete it after unsuccessful try of extracting the logic. And even missed it as I was self-reviewing the code 🤦🏼